bottlerocket/agents/src/bin/ec2-resource-agent/ec2_provider.rs

```diff
@@ -574,6 +582,100 @@ async fn get_instances_by_uuid(
         .collect())
 }
 
+/// Given a list of EC2 Instance ID, deregister their corresponding ECS container instances
+/// Note that we don't want to error if anything goes wrong here, we proceed with EC2 instance termination
+/// regardless of whether these calls succeed or not.
+async fn deregister_ecs_container_instances(
+    ecs_client: &aws_sdk_ecs::Client,
+    memo: &ProductionMemo,
+) {
+    let mut container_instances = Vec::new();
+    // Kick off ECS instance deregistration
+    for instance_id in &memo.instance_ids {
+        // Find the containter instance ID associated with the EC2 instance ID
+        if let Ok(list_output) = ecs_client
+            .list_container_instances()
+            .cluster(&memo.cluster_name)
+            .filter(format!("ec2InstanceId=={}", instance_id))
+            .send()
+            .await
+        {
+            // We expect a 1:1 mapping of EC2 instances to ECS container instances
+            if let Some(container_instance_id) = list_output
+                .container_instance_arns()
+                .and_then(|l| l.first())
+            {
+                // Store this container instance ID so we can check it later in a separate loop
+                container_instances.push(container_instance_id.to_string());
+                if let Err(e) = ecs_client
+                    .deregister_container_instance()
+                    .cluster(&memo.cluster_name)
+                    .container_instance(container_instance_id)
+                    .force(true)
+                    .send()
+                    .await
+                {
+                    warn!(
```

Comment:
Should this be `error!` instead?

Reply:
Since we're not erroring out on this, I think warn is appropriate. I usually consider error logs for messages that lead up to an abnormal exit of the program. In terms of logging destination, they both go to stderr.